nixos/modules/services/networking/routinator.nix

```
@@ -0,0 +1,71 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+with lib;
+let
+  cfg = config.services.routinator;
+  settingsFormat = pkgs.formats.toml { };
+in
+{
+  options.services.routinator = {
+    enable = mkEnableOption "Routinator 3000 is free, open-source RPKI Relying Party software made by NLnet Labs.";
+
+    settings = mkOption {
+      inherit (settingsFormat) type;
+      example = literalExpression ''
+
+        {
+          repository-dir = "/var/lib/routinator/rpki-cache";
+          rtr-listen = [ "127.0.0.1:3323" ];
+          http-listen = [ "127.0.0.1:8323" ];
+        }
+      '';
+      description = ''
+
+        Configuration for Routinator 3000, see <https://routinator.docs.nlnetlabs.nl/en/stable/manual-page.html#configuration-file> for options
+      '';
+      default = {
+        repository-dir = "/var/lib/routinator/rpki-cache";
+        rtr-listen = [ "127.0.0.1:3323" ];
+        http-listen = [ "127.0.0.1:8323" ];
+      };
+    };
+
+    configFile = mkOption {
+      default = settingsFormat.generate "routinator.conf" cfg.settings;
+      defaultText = "A Routinator configuration file automatically generated by values from services.routinator.settings.*";
+      type = types.path;
+      example = "/etc/routinator/routinator.conf";
+      description = "Path to Routinator TOML configuration file.";
+    };
+
+    package = mkPackageOption pkgs "routinator" { };
+  };
+
+  config = {
+    systemd.services.routinator = {
+      description = "Routinator 3000 is free, open-source RPKI Relying Party software made by NLnet Labs.";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      path = with pkgs; [ rsync ];
+      serviceConfig = {
+        ExecStart = "${lib.getExe cfg.package} --config ${cfg.configFile} server";
```

It is preferable to build commands as a list to ensure the correct tokenization and leave concatenation and escaping to a helper function.

Also, a great opportunity to enable an escape hatch for passing additional command line flags, think for example `--verbose`.

```suggestion
        ExecStart = lib.escapeSystemdExecArgs ([
        	(lib.getExe cfg.package)
        	"--config=${cfg.configFile}"
       ] ++ cfg.extraArgs ++ [
        	"server"
        ])
```

---

lib.escapeSystemdExecArgs does not exist, you probably mean utils.escapeSystemdExecArgs?